seastar/src/distributed.rs

```diff
@@ -100,15 +103,32 @@ where
     S: Service,
 {
     pub instance: &'a S,
-    pub container: &'a Distributed<S>,
+    pub container: &'a mut Distributed<S>,
+}
+
+pub struct PeeringShardedServiceMut<'a, S>
+where
+    S: Service,
+{
+    pub instance: &'a mut S,
+    pub container: &'a mut Distributed<S>,
 }
 
 /// A service distributed amongst all shards of a Seastar app.
 ///
 /// You can use this to, for example, load balance a local storage.
+/// One/many-to-one/many communication between instances of the service can be achieved
+/// via the use of `PeeringShardedService` or its `mut` equivalent, `PeeringShardedService`.
+///
+/// # Panics
+/// One can only delegeate a "mutating map" on an instance if all other "mutating maps" on it have ceased.
+/// In other words, you must ensure sequential acquisition of mutable borrows to an instance.
+/// Failing to comply with this rule will lead to a `panic!`, much in the same way that breaking the rule for
+/// a `RefCell` would do.
 pub struct Distributed<S: Service> {
     _inner: SharedPtr<distributed>,
     _ty: PhantomData<S>,
+    _locks: Vec<Arc<RwLock<()>>>,
```

Comment:
Seeing a synchronous `RwLock` here looks very scary because using synchronous locks in asynchronous code can easily lead to a deadlock. However, I understand that it was easier to use compared to `RefCell` because `RefCell` is not Sync/Send and it would cause lots of problems and the need to use unsafe. Please add a comment which explains what is the purpose of those locks.

Reply:
I believe this comment:
```
/// One can only delegate a "mutating map" on an instance if all other "mutating maps" on it have ceased.
/// In other words, you must ensure sequential acquisition of mutable borrows to an instance.
/// Failing to comply with this rule will lead to a `panic!`, much in the same way that breaking the rule for
/// a `RefCell` would do.
```
kind of explains it, but I'll add another one over the locks just in case.